components/SingleSelectRadio/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { ComponentContainer, RadioInput, RadioLabel } from './styles';
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface SingleSelectRadioGroupProps {
+  name: string;
+  options: Option[];
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+}
+
+export default function SingleSelectRadioGroup({
+  name,
+  options,
+  defaultValue,
+  onChange,
+}: SingleSelectRadioGroupProps) {
+  const [selectedValue, setSelectedValue] = useState(
+    defaultValue || options[0].value,
+  );
+
+  const handleChange = (value: string) => {
+    setSelectedValue(value);
+    if (onChange) onChange(value);
+  };
+
+  return (
+    <div>
+      {options.map(option => (
+        <ComponentContainer
+          key={option.value}
+          isSelected={selectedValue === option.value}
```

Comment:
we want the entire button to be clickable, so we can move the ` onChange={() => handleChange(option.value)}` into `<ComponentContainer>` (and make this a `styled.button` in `styles.ts`), and remove it from `<RadioInput>`